lib/src/widgets/movies_grid_view.dart

```diff
@@ -0,0 +1,101 @@
+
+class MoviesGridView extends StatefulWidget {
+  final Stream<Movie> allMovies;
+
+  const MoviesGridView({
+    Key? key,
+    required this.allMovies,
+  }) : super(key: key);
+
+  @override
+  _MoviesGridViewState createState() => _MoviesGridViewState();
+}
+
+class _MoviesGridViewState extends State<MoviesGridView> {
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder(
+      stream: widget.allMovies,
+      builder: (
+        context,
+        AsyncSnapshot<Movie> snapshot,
+      ) {
+        return snapshot.hasData
+            ? buildList(
+                snapshot,
+              )
+            : Center(
+                child: CircularProgressIndicator(
+                  backgroundColor: Colors.red,
+                  color: Colors.black,
+                ),
+              );
+      },
+    );
+  }
+}
+
+Widget buildList(AsyncSnapshot snapshot) {
+  return snapshot.hasData
+      ? GridView.builder(
+          itemCount: snapshot.data.results.length,
+          gridDelegate: new SliverGridDelegateWithFixedCrossAxisCount(
+            mainAxisExtent: MovieDimensions.mainAxisExtent,
+            crossAxisCount: MovieDimensions.gridColumns,
+          ),
+          itemBuilder: (
+            BuildContext context,
+            int index,
+          ) {
+            return GridTile(
+              child: Container(
+                margin: EdgeInsets.all(
+                  MovieDimensions.gridTileMargin,
+                ),
+                decoration: BoxDecoration(
+                  boxShadow: [
+                    BoxShadow(
+                      color: Colors.black.withOpacity(
+                        MovieDimensions.boxShadowOpacity,
+                      ),
+                      spreadRadius: MovieDimensions.boxShadowSpreadRadius,
+                      blurRadius: MovieDimensions.boxShadowBlurRadius,
+                    ),
+                  ],
+                ),
+                child: FadeIn(
+                  duration: Duration(
+                    seconds: MovieDimensions.fadeInPosterDuration,
+                  ),
+                  child: InkResponse(
+                    enableFeedback: true,
+                    child: Image.network(
+                      snapshot.data.results[index].posterPath,
+                      fit: BoxFit.cover,
+                    ),
+                    onTap: () {
+                      Navigator.push(
+                        context,
+                        MaterialPageRoute(
+                          builder: (
+                            context,
+                          ) =>
+                              MovieDetail(
+                            result: snapshot.data.results[index],
+                          ),
+                        ),
+                      );
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+        )
+      : Container();
```

Comment:
You can add some content to this Container, instead of an empty Container you can added an image or icon that show that something goes wrong when you doesn't have any data on the stream.

Reply:
movies_grid_view was refactored so Container is no longer available. Now you just consult one time if the stream hasData and create the buildList, other ways it just show the CircularProgressIndicator.